libkbfs/backpressure_disk_limiter.go

```diff
@@ -240,39 +245,46 @@ type backpressureDiskLimiter struct {
 	// actual semaphore itself.
 	lock                     sync.RWMutex
 	byteTracker, fileTracker *backpressureTracker
+	diskCacheByteTracker     *backpressureTracker
 }
 
-var _ diskLimiter = (*backpressureDiskLimiter)(nil)
+var _ DiskLimiter = (*backpressureDiskLimiter)(nil)
 
 // newBackpressureDiskLimiterWithFunctions constructs a new
 // backpressureDiskLimiter with the given parameters, and also the
 // given delay function, which is overridden in tests.
-func newBackpressureDiskLimiterWithFunctions(
-	log logger.Logger,
-	backpressureMinThreshold, backpressureMaxThreshold, limitFrac float64,
-	byteLimit, fileLimit int64, maxDelay time.Duration,
+func newBackpressureDiskLimiterWithFunctions(log logger.Logger,
+	backpressureMinThreshold, backpressureMaxThreshold, journalFrac,
+	diskCacheFrac float64, byteLimit, fileLimit int64, maxDelay time.Duration,
 	delayFn func(context.Context, time.Duration) error,
 	freeBytesAndFilesFn func() (int64, int64, error)) (
 	*backpressureDiskLimiter, error) {
 	freeBytes, freeFiles, err := freeBytesAndFilesFn()
 	if err != nil {
 		return nil, err
 	}
+	// byteLimit and fileLimit must be scaled by the proportion of the limit
+	// that the journal should consume.
+	journalByteLimit := int64((float64(byteLimit) * journalFrac) + 0.5)
 	byteTracker, err := newBackpressureTracker(
 		backpressureMinThreshold, backpressureMaxThreshold,
-		limitFrac, byteLimit, freeBytes)
+		journalFrac, journalByteLimit, freeBytes)
 	if err != nil {
 		return nil, err
 	}
+	journalFileLimit := int64((float64(fileLimit) * journalFrac) + 0.5)
```

Comment:
I don't think it makes sense to apply `journalFrac` to the file limit, since currently only the journal is using it. Maybe rename `journalFrac` and `diskCacheFrac` to `journalByteFrac`, etc. and apply those only to the byte trackers.